src/org/zaproxy/zap/extension/simpleExample/SimpleExampleAPI.java

```diff
@@ -0,0 +1,82 @@
+/*
+/* Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2016 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.simpleExample;
+
+import java.util.Locale;
+
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.api.API;
+import org.zaproxy.zap.extension.api.ApiAction;
+import org.zaproxy.zap.extension.api.ApiException;
+import org.zaproxy.zap.extension.api.ApiImplementor;
+import org.zaproxy.zap.extension.api.ApiResponse;
+import org.zaproxy.zap.extension.api.ApiResponseElement;
+
+import net.sf.json.JSONObject;
+
+public class SimpleExampleAPI extends ApiImplementor {
+    // TODO shouldnt allow unsafe-inline styles - need to work out where they are being used
+    protected static final String CSP_POLICY =
+        "default-src 'none'; script-src 'self'; connect-src https://zap wss://zap; frame-src 'self'; img-src 'self' data:; "
+        + "font-src 'self' data:; style-src 'self' 'unsafe-inline' ;";
+
+    private static final String PREFIX = "simpleExample";
+
+    private ExtensionSimpleExample extension;
+
+    private static final String ACTION_HELLO_WORLD = "helloWorld";
+
+    private static final Logger LOGGER = Logger.getLogger(SimpleExampleAPI.class);
+
+    public SimpleExampleAPI(ExtensionSimpleExample extension) {
+        this.extension = extension;
+        this.addApiAction(new ApiAction(ACTION_HELLO_WORLD));
+    }
+
+    @Override
+    public String getPrefix() {
+        return PREFIX;
+    }
+
+    @Override
+    public String handleCallBack(HttpMessage msg) throws ApiException {
```

Comment:
Remove as well

Reply:
I was kinda afraid to break things removing this one.
What is it used for?

Reply:
Thats the method that will be called if the add-on registers a callback with ZAP, ie a URL that is apparently on whatever target domain the add-on chooses :) In this case the add-on doesnt register for a callback, and I think its better it doesnt - thats too advanced for the intended use of this example and will only confuse people :)